cmd/gen-go-sample/out.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"strings"
+	"text/scanner"
+
+	"github.com/googleapis/gapic-generator-go/internal/errors"
+	"github.com/googleapis/gapic-generator-go/internal/pbinfo"
+)
+
+// symTab implements a symbol table.
+//
+// Some languages scope variables at function level. We track this with universe table.
+// Some languages scope variables to the curly-braces. We track this with parent and scope tables.
+//
+// So that our samples work with the first kind of languages, we only allow declaration
+// iff the name hasn't been declared in the universe scope.
+//
+// So that our samples work with the second kind of languages, we only allow variable references
+// in the recursive closure of the current scope.
+type symTab struct {
```

Comment:
Consider calling this `symbolTable`

Reply:
See above, `symtab` is a common abbreviation.